pipestat/backend.py

```diff
@@ -0,0 +1,1226 @@
+        table_name = self.get_table_name(pipeline_type=pipeline_type)
+        record_identifier = record_identifier or self.record_identifier
+        if rm_record:
+            try:
+                ORMClass = self.get_orm(table_name=table_name)
+                if self.check_record_exists(
+                    record_identifier=record_identifier, table_name=table_name
+                ):
+                    with self.session as s:
+                        records = s.query(ORMClass).filter(
+                            getattr(ORMClass, RECORD_ID) == record_identifier
+                        )
+                        records.delete()
+                        s.commit()
+                else:
+                    raise PipestatDatabaseError(f"Record '{record_identifier}' not found")
+            except Exception as e:
+                _LOGGER.error(f"Could not remove the result from the database. Exception: {e}")
+                raise
+        else:
+            _LOGGER.info(f" rm_record flag False, aborting Removing '{record_identifier}' record")
+
+    def get_model(self, table_name: str, strict: bool):
+        """
+        Get model based on table_name
+        :param str table_name: "sample" or "project"
+        :return model
+        """
+        orms = self.orms
+
+        # table_name = self.get_table_name()
+
+        mod = orms.get(table_name)
+
+        if strict and mod is None:
+            raise PipestatDatabaseError(
+                f"No object relational mapper class defined for table '{table_name}'. "
+                f"{len(orms)} defined: {', '.join(orms.keys())}"
+            )
+        return mod
+
+    def get_orm(self, table_name: str) -> Any:
+        """
+        Get an object relational mapper class
+
+        :param str table_name: table name to get a class for
+        :return Any: Object relational mapper class
+        """
+        if self.orms is None:
+            raise PipestatDatabaseError("Object relational mapper classes not defined")
+        mod = self.get_model(table_name=table_name, strict=True)
+        return mod
+
+    def get_one_record(self, table_name: str, rid: Optional[str] = None):
+        """
+        Retrieve single record from SQL table
+
+        :param str rid: record to check for
+        :param str table_name: table name to check
+        :return bool: whether the record exists in the table
+        """
+
+        models = [self.get_orm(table_name=table_name)] if table_name else list(self.orms.values())
+        with self.session as s:
+            for mod in models:
+                # record = sql_select(mod).where(mod.record_identifier == rid).first()
+                # record = s.query(mod).where(mod.record_identifier == rid).first()
+                stmt = sql_select(mod).where(mod.record_identifier == rid)
+                # stmt = sql_select(mod)
+                record = s.exec(stmt).first()
+                # record = (
```

Comment:
There are multiple places where code is commented. Can you delete this code, or make comments why it is commented.

Reply:
Done.